resources/views/components/select2.blade.php

```diff
@@ -1,22 +1,63 @@
-<div class="form-group {{$topclass}}">
-    <label for="{{$id}}">{{$label}}</label>
-    <select class="form-control {{$inputclass}} @error($name) is-invalid @enderror" 
-        id="{{$id}}" name="{{$name}}" style="width:100%" 
-        {{($required) ? 'required' : '' }} 
-        {{($disabled) ? 'disabled' : '' }}
-        {{($multiple) ? 'multiple' : '' }}>
-        {{$slot}}
+@extends('adminlte::components.input-group-component')
+
+@section('input_group_item')
+
+    {{-- Select --}}
+    <select id="{{ $name }}" name="{{ $name }}"
+        {{ $attributes->merge(['class' => $makeItemClass($errors->first($name))]) }}>
+        {{ $slot }}
     </select>
-    @error($name)
-        <span class="invalid-feedback" role="alert">
-            <strong>{{ $message }}</strong>
-        </span>
-    @enderror
-</div>
 
-@section('js')
-    @parent
+@overwrite
+
+{{-- Add plugin initialization and configuration code --}}
+
+@push('js')
     <script>
-        $(()=>{ $('#{{$id}}').select2({ theme: 'bootstrap4' }); })
+        $(() => {
+            $('#{{ $name }}').select2( @json($config) );
+        })
     </script>
-@endsection
\ No newline at end of file
+@endpush
+
+{{-- Setup the height and font size of the plugin when using sm/lg sizes --}}
+{{-- NOTE: this may change with newer plugin versions --}}
+
+@once
+@push('css')
```

Comment:
@REJack Do you know if `AdminLte` package is implementing this type of support on the incoming versions. Maybe these workarounds to support plugins inside and input group and work nicely with `input-group-[sm/lg]` can be added to the package in a future...

Reply:
There is nothing planed for select2 inside input-group but I'll look over it and may bring iz with v3.1. At the moment our custom select2 style has a support for form-control-sm.

Reply:
Thanks @REJack , maybe for the particular case of `select2` those customization may be inside [select2-bootstrap4-theme](https://github.com/ttskch/select2-bootstrap4-theme) plugin. However, I had to implement a similar fix for the `input-file` component, and probably I will have to do the same for other form components.